src/compiler/compile/nodes/Element.ts

```diff
@@ -351,6 +368,28 @@ export default class Element extends Node {
 					const match = fuzzymatch(value, aria_roles);
 					component.warn(attribute, compiler_warnings.a11y_unknown_role(value, match));
 				}
+
+				// no-redundant-roles
+				if (this.name === value) {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'nav' && value === 'navigation') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'a' && value === 'link') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'fieldset' && value === 'group') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				} else if (this.name === 'ul' && value === 'list') {
+					component.warn(attribute, compiler_warnings.a11y_no_redundant_roles(value));
+				}
```

Comment:
I just noticed that they are repeating `component.warn(...` lines. I think we can merge them into one, and have a big if condition for it, which might be more readable.

Reply:
It's not great to read I agree, maybe the conditions could be stored somewhere and a single check is out in place against those conditions? I can whip that up shortly.

Reply:
That could be a solution too, I'd personally keep it as an if condition so the logic is clear upfront. But I'm indifferent of either solutions, you can implement whichever you prefer 🙂

Reply:
You're right :) No need to seriously over-engineer things when readability could be preserved. `git commit`s incoming.

Reply:
Pushed in bb975e7